runtime/src/accounts_db.rs

```diff
@@ -6244,6 +6243,21 @@ impl AccountsDb {
     }
 
     pub fn store_cached(&self, slot: Slot, accounts: &[(&Pubkey, &AccountSharedData)]) {
+        let sum = self.accounts_cache.size();
+        if sum > 10_000_000_000 {
+            let changes: u64 = accounts.iter().map(|x| x.1.data().len() as u64).sum();
+            const MAX_DELAY_US: u64 = 10_000;
+            const BYTES_AT_MAX_DELAY: u64 = 10_000_000;
+            let delay = std::cmp::min(MAX_DELAY_US, MAX_DELAY_US * changes / BYTES_AT_MAX_DELAY);
+            datapoint_info!(
+                "accounts_db_store_cached_stall",
+                ("num_accounts", accounts.len(), i64),
+                ("delay_us", delay, i64),
+                ("bytes", changes, i64),
+            );
+            sleep(Duration::from_micros(delay));
```

Comment:
We don't have any way to enforce that this code is run on a block producer node. This means that a malicious leader can design a workload of transactions which backs up all the validators when they replay the malicious block. This is especially bad because replay stage is single threaded. So each transaction in a malicious batch results in a max sleep of 10ms and there is no limit to the number of transactions in a batch I believe.

Reply:
decided this should always run